File: src/main/java/com/baulsupp/oksocial/tracing/HttpUriHandler.java
package com.baulsupp.oksocial.tracing;

import java.net.URI;
import java.util.Optional;
import zipkin.reporter.Sender;
import zipkin.reporter.okhttp3.OkHttpSender;

import static java.util.Optional.empty;
import static java.util.Optional.of;

public class HttpUriHandler implements UriHandler {
  @Override public Optional<Sender> buildSender(URI uri) {
    if (uri.getScheme().equals("http") || uri.getScheme().equals("https")) {
      return of(OkHttpSender.create(uri.toString()));
    }
    return empty();
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/tracing/HttpUriHandler.kt
package com.baulsupp.oksocial.tracing

import zipkin2.Span
import zipkin2.reporter.AsyncReporter
import zipkin2.reporter.Reporter
import zipkin2.reporter.okhttp3.OkHttpSender
import java.net.URI

class HttpUriHandler : UriHandler {
  override fun buildSender(uri: URI): Reporter<Span>? = when {
    uri.scheme == "http" || uri.scheme == "https" -> AsyncReporter.create(OkHttpSender.create(uri.toString()))
    else -> null
  }
}
